src/components/auth/login.js

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { LoginContext } from '../../context/auth-context.js';
+
+const If = props => {
+  return props.condition ? props.children : null;
+};
+
+class Login extends React.Component {
+  static contextType = LoginContext;
+
+  constructor(props) {
+    super(props);
+    this.state = { username: '', password: '' };
+  }
+
+  handleChange = e => {
+    this.setState({ [e.target.name]: e.target.value });
+  };
+
+  handleSubmit = e => {
+    e.preventDefault();
+    this.context.login(this.state.username, this.state.password);
+  };
+
+  render() {
+    return (
+      <>
+        <If condition={this.context.loggedIn}>
+          <button onClick={this.context.logout}>Log Out</button>
+        </If>
+
+        <If condition={!this.context.loggedIn}>
+          <form onSubmit={this.handleSubmit}>
+            <input
+              placeholder="UserName"
+              name="username"
+              onChange={this.handleChange}
+            />
+            <input
+              placeholder="password"
+              name="password"
+              onChange={this.handleChange}
```

Comment:
If you want to do a little extra here, you can add `type = "password"` and that will make those dots show up instead of seeing the password.